src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -201,14 +201,14 @@ private CSharpCompilation Rewrite(CSharpCompilation compilation)
 
         // Internal for unit testing
         internal CompilationResult GetCompilationFailedResult(
-            string relativePath,
+            RelativeFileInfo fileInfo,
             string compilationContent,
             string assemblyName,
             IEnumerable<Diagnostic> diagnostics)
         {
             var diagnosticGroups = diagnostics
                 .Where(IsError)
-                .GroupBy(diagnostic => GetFilePath(relativePath, diagnostic), StringComparer.Ordinal);
+                .GroupBy(diagnostic => GetFilePath(fileInfo.RelativePath, diagnostic), StringComparer.Ordinal);
```

Comment:
Having the developer exception page in production is pretty bad. AFAIK, the database error page lets you scaffold too, so you'd be in pretty bad shape.